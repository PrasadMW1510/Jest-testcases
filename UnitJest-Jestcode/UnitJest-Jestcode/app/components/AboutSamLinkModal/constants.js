const SAM_SERVER_NAME = 'SAM Server';
export default SAM_SERVER_NAME;
