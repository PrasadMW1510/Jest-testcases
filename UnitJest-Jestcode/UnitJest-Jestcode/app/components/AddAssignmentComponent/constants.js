export const CLASS_SCORE = 'score';
export const ASSIGNMENT_NAME_ERROR = '.Please select a Fluency Passage';
