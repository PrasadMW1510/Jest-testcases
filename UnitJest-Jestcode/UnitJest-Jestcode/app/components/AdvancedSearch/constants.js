/*
 *
 * AdvanedSearch constants
 *
 */

export const BOOK_LEXILE_RANGE = 'This value may not exceed 1700.';
export const BOOK_READING_RANGE = 'This value may not exceed 12.9';
export const BOOK_POINTS_RANGE = 'This value may not exceed 9999';
export const BOOK_LEXILE_MAX_RANGE = 'Maximum value required for the Lexile Range(Range:0-1700)';
export const BOOK_LEXILE_MIN_RANGE = 'Minimum value required for the Lexile Range(Range:0-1700)';
export const BOOK_READING_MAX_RANGE =
  'Maximum value required for the Reading Level Range(Range:0-12.9)';
export const BOOK_READING_MIN_RANGE =
  'Minimum value required for the Reading Level Range(Range:0-12.9)';
export const BOOK_GUIDEREADING_MAX =
  ' Maximum value required for the Guided Reading Level Range(Range:A-Z)';
export const BOOK_GUIDEREADING_MIN =
  'Minimum value required for the Guided Reading Level Range(Range:A-Z)';
export const BOOK_POINTS_RANGE_MAX =
  'Maximum value required for the Points Range Level Range(Range:0-9999)';
export const BOOK_POINTS_RANGE_MIN =
  'Minimum value required for the Points Range Level Range(Range:0-9999)';
export const BOOK_LEXILE_MAX_RANGE_VALUE = 1700;
export const BOOK_READING_RANGE_VALUE = 12.9;
export const BOOK_POINTS_RANGE_VALUE = 9999;
export const INDEX_NEG = -1;
export const INDEX_POS = 1;
export const BOOK_LEXILE_MAX_MIN = 'A Minimum value may not be greater than maximum value';
