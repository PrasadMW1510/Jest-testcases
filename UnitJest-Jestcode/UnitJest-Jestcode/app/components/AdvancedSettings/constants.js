export const EXPORT_URL_PATH = 'export.s';
export const IMPORT_URL_PATH = 'import.s';
export const STUDENT_PROMOTIONS_URL_PATH = 'endOfYear.s';
export const VIEW_OUTPUT_FILES_URL_PATH = 'fileList.s';
