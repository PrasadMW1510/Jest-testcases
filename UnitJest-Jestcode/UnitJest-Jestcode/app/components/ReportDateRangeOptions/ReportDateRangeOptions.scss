.time-period-box {
  height: 151px;
}

.report-settings-box-description {
  height: 126px;
}

.report-date-range-option-value {
  font-family: sans-serif;
}
