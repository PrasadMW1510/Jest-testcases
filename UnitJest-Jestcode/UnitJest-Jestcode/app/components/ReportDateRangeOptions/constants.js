const dateRangeText = {
  Today: 'Today',
  ThisWeek: 'This Week',
  LastWeek: 'Last Week',
  CurrSchoolYear: 'This School Year',
  CurrGradingPeriod: 'This Grading Period',
  GradingPeriod1: 'Grading Period 1',
  GradingPeriod2: 'Grading Period 2',
  GradingPeriod3: 'Grading Period 3',
  GradingPeriod4: 'Grading Period 4',
  GradingPeriod5: 'Grading Period 5',
  GradingPeriod6: 'Grading Period 6',
  Last3Months: 'Last 3 Months',
  Custom: 'Custom',
  StartOfOperation: 'Start - Today',
  SinceStartDate: 'All Dates',
  Last10Days: 'Last 10 Days',
  Last15Days: 'Last 15 Days',
  Last30Days: 'Last 30 Days',
};

export default dateRangeText;
