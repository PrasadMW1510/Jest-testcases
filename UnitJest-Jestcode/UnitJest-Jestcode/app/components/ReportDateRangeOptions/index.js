/**
 *
 * ReportDateRangeOptions
 *
 */

export { default } from './ReportDateRangeOptions';
