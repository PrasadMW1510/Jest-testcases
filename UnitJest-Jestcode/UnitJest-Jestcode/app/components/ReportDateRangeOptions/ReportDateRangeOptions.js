/**
 *
 * ReportDateRangeOptions
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import * as AppConstants from 'containers/App/constants';
import './ReportDateRangeOptions.scss';
import dateRangeText from './constants';

class ReportDateRangeOptions extends React.Component {
  // eslint-disable-line react/prefer-stateless-function

  selected(index) {
    return index === 0;
  }

  render() {
    if (typeof this.props.dateRanges === 'undefined') {
      return (
        <div id="TimePeriodBox" className="report-right-box">
          <div className="report-settings-box-title">Time Period</div>
        </div>
      );
    }
    return (
      <div className="time-period-box report-right-box">
        <div className="report-settings-box-title">Time Period</div>
        <div className="report-settings-box report-settings-box-description" id="ReportDateRanges">
          {this.props.dateRanges.map(
            (dateRange, index) =>
              this.props.cohortType === AppConstants.COHORT_TYPE.District &&
              dateRange.match(/Grading/) ? (
                ''
              ) : (
                <div className="report-date-range-option" key={dateRange}>
                  <label htmlFor="ReportDateRange" className="report-date-range-option-value">
                    <input
                      type="radio"
                      name="ReportDateRange"
                      className="report-date-range-radio"
                      value={dateRange}
                      onClick={this.props.chooseDate}
                      defaultChecked={this.selected(index)}
                    />
                    {dateRangeText[dateRange]}
                  </label>
                </div>
              )
          )}
        </div>
      </div>
    );
  }
}

ReportDateRangeOptions.propTypes = {
  dateRanges: PropTypes.array,
  chooseDate: PropTypes.func.isRequired,
  cohortType: PropTypes.string,
};

export default ReportDateRangeOptions;
