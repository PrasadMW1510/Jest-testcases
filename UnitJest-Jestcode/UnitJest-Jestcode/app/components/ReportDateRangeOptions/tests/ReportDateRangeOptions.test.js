import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import * as AppConstants from 'containers/App/constants';
import ReportDateRangeOptions from '../index';

describe('<ReportDateRangeOptions />', () => {
  let wrapper = null;
  let dateRanges = null;
  let chooseDate = null;
  let cohortType = null;
  beforeEach(() => {
    dateRanges = [
      'CurrSchoolYear',
      'CurrGradingPeriod',
      'GradingPeriod1',
      'GradingPeriod2',
      'GradingPeriod3',
      'GradingPeriod4',
      'GradingPeriod5',
      'GradingPeriod6',
      'Custom',
    ];
    chooseDate = jest.fn();
    cohortType = AppConstants.COHORT_TYPE.School;
    wrapper = shallow(
      <ReportDateRangeOptions
        dateRanges={dateRanges}
        chooseDate={chooseDate}
        cohortType={cohortType}
      />
    );
  });

  it('Expect to render correctly', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('Expect to render correctly with no date range options', () => {
    dateRanges = undefined;
    wrapper.setProps({ dateRanges });
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('Expect to render correctly with district cohort type', () => {
    cohortType = AppConstants.COHORT_TYPE.District;
    wrapper.setProps({ cohortType });
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('Expect to handle (non-custom) radio button clicks', () => {
    wrapper
      .find('.report-date-range-radio')
      .first()
      .simulate('click', { target: { className: 'report-date-range-radio', value: 'Today' } });
    expect(chooseDate.mock.calls.length).toBe(1);
  });
});
