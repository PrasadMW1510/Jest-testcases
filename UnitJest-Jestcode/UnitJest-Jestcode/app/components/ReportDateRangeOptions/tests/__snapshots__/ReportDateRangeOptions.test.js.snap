// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReportDateRangeOptions /> Expect to render correctly 1`] = `
<div
  className="time-period-box report-right-box"
>
  <div
    className="report-settings-box-title"
  >
    Time Period
  </div>
  <div
    className="report-settings-box report-settings-box-description"
    id="ReportDateRanges"
  >
    <div
      className="report-date-range-option"
      key="CurrSchoolYear"
    >
      <label
        className="report-date-range-option-value"
        htmlFor="ReportDateRange"
      >
        <input
          className="report-date-range-radio"
          defaultChecked={true}
          name="ReportDateRange"
          onClick={[Function]}
          type="radio"
          value="CurrSchoolYear"
        />
        This School Year
      </label>
    </div>
    <div
      className="report-date-range-option"
      key="CurrGradingPeriod"
    >
      <label
        className="report-date-range-option-value"
        htmlFor="ReportDateRange"
      >
        <input
          className="report-date-range-radio"
          defaultChecked={false}
          name="ReportDateRange"
          onClick={[Function]}
          type="radio"
          value="CurrGradingPeriod"
        />
        This Grading Period
      </label>
    </div>
    <div
      className="report-date-range-option"
      key="GradingPeriod1"
    >
      <label
        className="report-date-range-option-value"
        htmlFor="ReportDateRange"
      >
        <input
          className="report-date-range-radio"
          defaultChecked={false}
          name="ReportDateRange"
          onClick={[Function]}
          type="radio"
          value="GradingPeriod1"
        />
        Grading Period 1
      </label>
    </div>
    <div
      className="report-date-range-option"
      key="GradingPeriod2"
    >
      <label
        className="report-date-range-option-value"
        htmlFor="ReportDateRange"
      >
        <input
          className="report-date-range-radio"
          defaultChecked={false}
          name="ReportDateRange"
          onClick={[Function]}
          type="radio"
          value="GradingPeriod2"
        />
        Grading Period 2
      </label>
    </div>
    <div
      className="report-date-range-option"
      key="GradingPeriod3"
    >
      <label
        className="report-date-range-option-value"
        htmlFor="ReportDateRange"
      >
        <input
          className="report-date-range-radio"
          defaultChecked={false}
          name="ReportDateRange"
          onClick={[Function]}
          type="radio"
          value="GradingPeriod3"
        />
        Grading Period 3
      </label>
    </div>
    <div
      className="report-date-range-option"
      key="GradingPeriod4"
    >
      <label
        className="report-date-range-option-value"
        htmlFor="ReportDateRange"
      >
        <input
          className="report-date-range-radio"
          defaultChecked={false}
          name="ReportDateRange"
          onClick={[Function]}
          type="radio"
          value="GradingPeriod4"
        />
        Grading Period 4
      </label>
    </div>
    <div
      className="report-date-range-option"
      key="GradingPeriod5"
    >
      <label
        className="report-date-range-option-value"
        htmlFor="ReportDateRange"
      >
        <input
          className="report-date-range-radio"
          defaultChecked={false}
          name="ReportDateRange"
          onClick={[Function]}
          type="radio"
          value="GradingPeriod5"
        />
        Grading Period 5
      </label>
    </div>
    <div
      className="report-date-range-option"
      key="GradingPeriod6"
    >
      <label
        className="report-date-range-option-value"
        htmlFor="ReportDateRange"
      >
        <input
          className="report-date-range-radio"
          defaultChecked={false}
          name="ReportDateRange"
          onClick={[Function]}
          type="radio"
          value="GradingPeriod6"
        />
        Grading Period 6
      </label>
    </div>
    <div
      className="report-date-range-option"
      key="Custom"
    >
      <label
        className="report-date-range-option-value"
        htmlFor="ReportDateRange"
      >
        <input
          className="report-date-range-radio"
          defaultChecked={false}
          name="ReportDateRange"
          onClick={[Function]}
          type="radio"
          value="Custom"
        />
        Custom
      </label>
    </div>
  </div>
</div>
`;

exports[`<ReportDateRangeOptions /> Expect to render correctly with district cohort type 1`] = `
<div
  className="time-period-box report-right-box"
>
  <div
    className="report-settings-box-title"
  >
    Time Period
  </div>
  <div
    className="report-settings-box report-settings-box-description"
    id="ReportDateRanges"
  >
    <div
      className="report-date-range-option"
      key="CurrSchoolYear"
    >
      <label
        className="report-date-range-option-value"
        htmlFor="ReportDateRange"
      >
        <input
          className="report-date-range-radio"
          defaultChecked={true}
          name="ReportDateRange"
          onClick={[Function]}
          type="radio"
          value="CurrSchoolYear"
        />
        This School Year
      </label>
    </div>
    <div
      className="report-date-range-option"
      key="Custom"
    >
      <label
        className="report-date-range-option-value"
        htmlFor="ReportDateRange"
      >
        <input
          className="report-date-range-radio"
          defaultChecked={false}
          name="ReportDateRange"
          onClick={[Function]}
          type="radio"
          value="Custom"
        />
        Custom
      </label>
    </div>
  </div>
</div>
`;

exports[`<ReportDateRangeOptions /> Expect to render correctly with no date range options 1`] = `
<div
  className="report-right-box"
  id="TimePeriodBox"
>
  <div
    className="report-settings-box-title"
  >
    Time Period
  </div>
</div>
`;
